tests/vcs/VCSCommitBearTest.py

```diff
@@ -0,0 +1,179 @@
+        return True
+
+    def get_head_commit(self):
+        return ('This is the fake head commit', '')
+
+
+@generate_skip_decorator(VCSCommitBear)
+class VCSCommitBearTest(unittest.TestCase):
+
+    def run_uut(self, *args, **kwargs):
+        """
+        Runs the unit-under-test (via `self.uut.run()`) and collects the
+        messages of the yielded results as a list.
+
+        :param args:   Positional arguments to forward to the run function.
+        :param kwargs: Keyword arguments to forward to the run function.
+        :return:       A list of the message strings.
+        """
+        return list(result.message for result in self.uut.run(*args, **kwargs))
+
+    def assert_no_msgs(self):
+        """
+        Assert that there are no messages in the message queue of the bear, and
+        show the messages in the failure message if it is not empty.
+        """
+        self.assertTrue(
+            self.msg_queue.empty(),
+            'Expected no messages in bear message queue, but got: ' +
+            str(list(str(i) for i in self.msg_queue.queue)))
+
+    def setUp(self):
+        self.msg_queue = Queue()
+        self.section = Section('')
+        self.uut = FakeCommitBear(None, self.section, self.msg_queue)
+
+    def test_check_prerequisites(self):
+        _shutil_which = shutil.which
+        try:
+            shutil.which = lambda *args, **kwargs: None
+            self.assertEqual(GitCommitBear.check_prerequisites(),
+                             'git is not installed.')
+
+            shutil.which = lambda *args, **kwargs: 'path/to/git'
+            self.assertTrue(GitCommitBear.check_prerequisites())
+        finally:
+            shutil.which = _shutil_which
+
+    def test_head_commit(self):
+        self.assertEqual(self.run_uut(), ['HEAD commit information'])
+
+
+@generate_skip_decorator(GitCommitBear)
+class AnalyzeGitCommitTest(unittest.TestCase):
+
+    @staticmethod
+    def run_git_command(*args, stdin=None):
+        run_shell_command(' '.join(('git',) + args), stdin)
+
+    @staticmethod
+    def git_commit(msg):
+        # Use stdin mode from git, since -m on Windows cmd does not support
+        # multiline messages.
+        AnalyzeGitCommitTest.run_git_command('commit',
+                                             '--allow-empty',
+                                             '--allow-empty-message',
+                                             '--file=-',
+                                             stdin=msg)
+
+    def run_uut(self, *args, **kwargs):
+        """
+        Runs the unit-under-test (via `self.uut.run()`) and collects the
+        messages of the yielded results as a list.
+
+        :param args:   Positional arguments to forward to the run function.
+        :param kwargs: Keyword arguments to forward to the run function.
+        :return:       A list of the message strings.
+        """
+        return list(result.message for result in self.uut.run(*args, **kwargs))
```

Comment:
Return all of the yielded values from `uut.run()`, not only the message.
`return list(self.uut.run(*args, **kwargs))

Reply:
This also generates an output like 
`
[<CommitResult object(id=0xa4641b38563489790f5f40e78ef1ff3, origin='GitCommitBear', commit_info=(<class 'bears.vcs.VCSCommitBear.Commit_Info'>,), message='HEAD commit information') at 0x7f311e4a30f0>]
`

checking which is similar to https://github.com/coala/coala-bears/pull/2543/files/8484120bc877ab722a9fad4e3bc2f4e9706d43ca#diff-0704b7478b1a43f7ae7b1f2cafe0d8c3R170

Should I still do it this way ?
 Ideally we should check the contents of `<class 'bears.vcs.VCSCommitBear.Commit_Info'>`

Reply:
Yes, that's a `CommitResult` object yielded by `VCSBear`. It's intended

You can make an instance of `CommitBear` that you're expecting by yourself in the unittest
```py
        self.assertEqual(self.uut.run(), [CommitResult(self.uut, commit_info)]
``